update_agol_vehicles_pallet.py

```diff
@@ -0,0 +1,201 @@
+        sd_path = sddraft_path[:-5]
+
+        paths = [temp_csv_dir, temp_fc_path, sddraft_path, sd_path]
+        for item in paths:
+            if arcpy.Exists(item):
+                self.log.info(f'Deleting {item} prior to use...')
+                arcpy.Delete_management(item)
+        os.mkdir(temp_csv_dir)
+
+        if not secrets.KNOWNHOSTS or not os.path.isfile(secrets.KNOWNHOSTS):
+            raise FileNotFoundError(
+                f'known_hosts file {secrets.KNOWNHOSTS} not found. Please '
+                'create with ssh-keyscan.')
+
+        #: Download all the files in the upload folder on sftp to temp_csv_dir
+        self.log.info(
+            f'Downloading all files from {secrets.SFTP_HOST}/upload...')
+        connection_opts = pysftp.CnOpts(knownhosts=secrets.KNOWNHOSTS)
+        with pysftp.Connection(
+                secrets.SFTP_HOST, username=secrets.SFTP_USERNAME,
+                password=secrets.SFTP_PASSWORD, cnopts=connection_opts) as sftp:
+            sftp.get_d('upload', temp_csv_dir, preserve_mtime=True)
+
+        #: Get the latest file
+        source_path_object, source_date = get_latest_csv(
+            temp_csv_dir, self.log, previous_days=7)
+        source_path = str(source_path_object)
+
+        self.log.info(
+            f'Converting {source_path} to feature class {temp_fc_path}...')
+        wgs84 = arcpy.SpatialReference(4326)
+        result = arcpy.management.XYTableToPoint(
+            source_path, temp_fc_path, 'LONGITUDE', 'LATITUDE',
+            coordinate_system=wgs84)
+        self.log.debug(result.getMessages())
+
+        try:
+            #: Overwrite existing AGOL service
+            self.log.info(f'Connecting to AGOL as {secrets.AGOL_USERNAME}...')
+            gis = arcgis.gis.GIS(
+                'https://www.arcgis.com', secrets.AGOL_USERNAME,
+                secrets.AGOL_PASSWORD)
+            sd_item = gis.content.get(secrets.SD_ITEM_ID)
+
+            self.log.info('Getting map and layer...')
+            layer, fleet_map = get_map_layer(
+                secrets.PROJECT_PATH, temp_fc_path, self.log)
+
+            #: draft, stage, update, publish
+            self.log.info(f'Staging and updating...')
+            update_agol_feature_service(
+                fleet_map, layer, feature_service_name, sddraft_path,
+                sd_path, sd_item)
+
+            #: Update item description
+            self.log.info('Updating item description...')
+            feature_item = gis.content.get(secrets.FEATURES_ITEM_ID)
+            description = ('Vehicle location data obtained from Fleet; '
+                           f'updated on {source_date}')
+            feature_item.update(item_properties={'description': description})
+        except HTTPError as e:
+            self.log.info(f'Connection error with {e.url}, probably related to '
+                          'connection with AGOL.')
+            raise e
```

Comment:
**thought**: should this have a basic [retry policy](https://github.com/agrc/geocoding-toolbox/blob/master/src/agrcgeocoding/geocode.py#L83-L107)?

Reply:
Will now sleep for 10 seconds after a HTTPError or URLError, retrying twice.